docs/pages/enroll-resources/mcp-access/integration-guides/vault.mdx

```diff
@@ -0,0 +1,121 @@
+
+## Prerequisites
+
+(!docs/pages/includes/edition-prereqs-tabs.mdx edition="Teleport (v18.3.0 or higher)" clients="\`tsh\` client"!)
+- Access to your Grafana instance and sufficient privileges to manage service accounts.
+- A host to run the MCP server that is reachable by the Teleport Application Service.
+- Running [Teleport Application Service](../getting-started.mdx).
+- A Teleport user with sufficient permissions (e.g. role `mcp-user`) to access
+  MCP servers.
+
+## Step 1/3. Create a service in Vault
+
+First, create a policy file:
+```code
+$ cat > mcp-readonly.hcl <<EOF
+# Read/list all secrets and metadata stored in the KV v2 engine at "secret/"
+path "secret/data/*" {
+  capabilities = ["read", "list"]
+}
+path "secret/metadata/*" {
+  capabilities = ["read", "list"]
+}
+EOF
+```
+This example grants read-only access to all secrets stored under the `secret/` KV
+v2 engine. You can tighten or expand these paths depending on your needs.
+
+To load the policy into Vault:
+```code
+$ vault policy write mcp-readonly mcp-readonly.hcl
+```
+
+Once the policy created, generate a Vault token that the MCP server will use:
+```code
+$ vault token create -policy="mcp-readonly" -display-name="teleport-mcp-service" -ttl=720h
+```
+
+Copy the result token for use in the next step.
+
+## Step 2/3. Run the Vault MCP server
+
+The [Vault MCP Server](https://github.com/hashicorp/vault-mcp-server) can be run
+either as a compiled binary or via the official Docker image:
+
+<Tabs>
+<TabItem label="vault-mcp-server binary">
+To start the MCP server in streamable-HTTP mode:
+```code
+$ export TRANSPORT_MODE=http
+$ export TRANSPORT_HOST=<Var name="MCP_HOST" initial="localhost" /> # or listen to a network that is reachable by Teleport
+$ export VAULT_ADDR=<Var name="VAULT_ADDR" initial_value="https://your-vault-address:8200" />
+$ export VAULT_TOKEN=<Var name="VAULT_TOKEN" initial_value="your_vault_token"/>
+$ ./vault-mcp-server
+```
+
+Replace `<Var name="MCP_HOST"/>` with the hostname of the host machine running
+the MCP server. The host must be reachable by the Teleport Application
+Service.
+</TabItem>
+
+<TabItem label="docker">
+To start the MCP server in streamable-HTTP mode:
+```code
+$ docker run -d -p 8080:8080 \
+  -e TRANSPORT_MODE=http \
+  -e TRANSPORT_HOST=0.0.0.0 \
+  -e VAULT_ADDR=<Var name="VAULT_ADDR" initial_value="https://your-vault-address:8200" /> \
+  -e VAULT_TOKEN=<Var name="VAULT_TOKEN" initial_value="your_vault_token"/> \
+  --name vault-teleport-mcp hashicorp/vault-mcp-server
+```
+</TabItem>
+</Tabs>
+
+After starting, the Vault MCP Server exposes a streamable-HTTP endpoint at
+`http://<Var name="MCP_HOST" initial="localhost" />:8080/mcp`.
+
+## Step 3/3. Connect via Teleport
+
+(!docs/pages/includes/mcp-access/integration-teleport-app.mdx service="vault"  serviceName="Vault" port="8080" !)
+
+<Admonition type="tip" title="Limiting MCP Tools">
```

Comment:
For me, this admonition is too long. On my display, for example, it's about a third of the client height, and becomes less of an aside and more of a substantial part of the content. I would make this part of the main flow of the instructions instead, albeit an optional step.